torch/distributed/_sharded_tensor/ops/_common.py

```diff
@@ -124,14 +127,20 @@ def _result_distribute_with_col_rearrange(
     # Compute output splits
     split_size = get_split_size(sharding_dim_size, world_size)
     output_split_sizes = [0] * world_size
+    output_tensor_list = [None] * world_size
     for idx, placement in enumerate(weight._sharding_spec.placements):
-        output_split_sizes[placement.rank()] = get_chunked_dim_size(
-            sharding_dim_size, split_size, idx
+        output_split_size = get_chunked_dim_size(sharding_dim_size, split_size, idx)
+        output_split_sizes[placement.rank()] = output_split_size
+        dims[0] = output_split_size
+        output_tensor_list[placement.rank()] = torch.empty(
+            *dims,
+            device=input.device,
+            dtype=combined_results.dtype,
```

Comment:
Can't we remove `combined_results` completely and avoid the `torch.cat` above? I'm assuming you can just use `results[0].dtype` here?

Reply:
Opps, we definitely need to remove it.